scss/_patterns_lists.scss

```diff
@@ -199,19 +199,64 @@ $list-step-bullet-margin: $sph--x-large;
   }
 }
 
+// todo this extraction may not be necessary but keeping it for now should the list items need more styling
+@mixin list-horizontal-item-intrinsic-sizing-impl($num-cols) {
+  grid-column-end: span #{$num-cols};
+}
+
+/**
+ * Applies intrinsic sizing to horizontal list items by adjusting their grid column span based on the container's width.
+ * Requires that the parent container has `container-type` set to `inline-size` or `size`.
+ * @param {number} $min-width - The min width that the entire list should span, not including gutter adjustments
+ * @param {number} $num-cols - The number of columns the list item should span.
+ * @param {number} $gutter-width - The width of the gutter (space) between grid items. This is used to calculate the required width for the item.
+ *
+ * Example Usage:
+ * ```scss
+ * <selector> {
+ *   @include list-horizontal-item-intrinsic-sizing(30, 4, 1rem);
+ * }
+ * ```
+ */
+@mixin list-horizontal-item-intrinsic-sizing($min-width, $num-cols, $gutter-width) {
+  @if $min-width {
+    // Constrain to min width, taking gutters into account
+    @container (width > #{$min-width - ($num-cols * $gutter-width)}) {
+      @include list-horizontal-item-intrinsic-sizing-impl($num-cols);
+    }
+  } @else {
+    @include list-horizontal-item-intrinsic-sizing-impl($num-cols);
+  }
+}
+
+@mixin vf-p-list-horizontal {
+  .p-list--horizontal {
+    @extend %vf-list-with-state-icons;
+    @extend %vf-grid-row;
+    container-type: inline-size;
+  }
+  .p-list--horizontal .p-list__item {
+    @extend %vf-list-item;
+    border-top: $border;
+  }
+
+  .p-list--horizontal.is-4-cols-on-large .p-list__item {
+    // note that the 4/4/8 grid will snap at different breakpoints than these list items
+    @include list-horizontal-item-intrinsic-sizing(null, $grid-8-columns-small, map-get($grid-gutter-widths, small));
```

Comment:
Passing "null" as first param seems a bit dirty. Maybe it can be turned into optional param with default value of null?

Reply:
Resolved to no longer use this approach at all